core/storage/email/gae_models.py

```diff
@@ -80,6 +80,24 @@ def get_deletion_policy():
         """Sent email should be kept for audit purposes."""
         return base_models.DELETION_POLICY.KEEP
 
+    @classmethod
+    def has_reference_to_user_id(cls, user_id):
+        """Check whether SentEmailModel exists for user.
+
+        Args:
+            user_id: str. The ID of the user whose data should be checked.
+
+        Returns:
+            bool. Whether any models refer to the given user ID.
+        """
+        references_recipient_id = cls.query(
+            cls.recipient_id == user_id
+        ).get() is not None
+        references_sender_id = cls.query(
```

Comment:
Can we do this query in one call?

Reply:
No, there is no support for ORs in queries.

Reply:
I think we can do AND and OR operations as documented [here](https://cloud.google.com/appengine/docs/standard/python/ndb/queries#nest_and_or).

Reply:
:open_mouth: I previously used `.filter` and it didn't provide that and I didn't realize query might allow more commands, thanks.

Reply:
Done.